### PR TITLE
fix(form): success feedback should display when pass hasFeedback prop and current value is valid value

### LINKED ISSUE #41084: Form feedback icon not showing if the input has valid value while using validateFields method
### Reproduction link

[https://stackblitz.com/edit/react-nspdpm?file=demo.tsx](https://stackblitz.com/edit/react-nspdpm?file=demo.tsx)

### Steps to reproduce

1. Run the stackblitz demo
2. Check the input boxes with valid and invalid email values.
3. Feedback shows for invalid input only

### What is expected?

I'm expecting feedback icon (success/error) should show for both valid or invalid values while validating the form if the Form.Item has 'hasFeedback' prop as true

### What is actually happening?

error feedback icon only showing while using validateFields method, success feedback not showing in input while using the validateFields method. Success feedback icon rendering when start changing the value in input.

| Environment | Info |
| --- | --- |
| antd | 4.24.8 |
| React | 17 |
| System | Stackblitz |
| Browser | Chrome |

### PR DESCRIPTION
**Pull request by kiner-tang**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [X] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

- fix: https://github.com/ant-design/ant-design/issues/41084

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | success feedback should display when pass hasFeedback prop and current value is valid value |
| 🇨🇳 Chinese | 当显式传递`hasFeedback`属性且当前值是合法值时应展示成功反馈 |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [X] Doc is updated/provided or not needed
- [X] Demo is updated/provided or not needed
- [X] TypeScript definition is updated/provided or not needed
- [X] Changelog is provided or not needed

### CODE REVIEW

File: components/form/FormItem/ItemHolder.tsx [hunk 1]
```diff
@@ -90,15 +90,16 @@ export default function ItemHolder(props: ItemHolderProps) {
     mergedValidateStatus = 'error';
   } else if (debounceWarnings.length) {
     mergedValidateStatus = 'warning';
-  } else if (meta.touched) {
+  } else if (meta.touched || hasFeedback) {
```
Comment (zombieJ) [inline components/form/FormItem/ItemHolder.tsx:93]: 感觉不太对，如果 validateTrigger 是 onBlur，这里可能没有触发过校验。

Comment (zombieJ) [reply components/form/FormItem/ItemHolder.tsx:93]: 原本的逻辑里感觉就有这个 bug

Comment (kiner-tang) [reply components/form/FormItem/ItemHolder.tsx]: done

File: components/form/demo/validate-static.tsx [hunk 2]
```diff
@@ -25,125 +26,154 @@ const formItemLayout = {
...
+      <Form.Item label="Fail" validateStatus="error" hasFeedback>
+        <Mentions />
+      </Form.Item>
+
+      <Form.Item label="Fail" validateStatus="error" hasFeedback help="Should have something">
+        <Input.TextArea allowClear showCount />
+      </Form.Item>
+    </Form>
+  );
+};
```
[the remark this answers is not in the document]
Comment (kiner-tang) [reply components/form/demo/validate-static.tsx]: 我们本次更改已经有测试用例覆盖了，这个示例可以不用了，我删掉吧

**Review by zombieJ — APPROVED** [review]

**Pull request merged**